rasa/core/training/story_reader/story_reader.py

```diff
@@ -42,12 +42,24 @@ def _new_story_part(self, name: Text, source_name: Text):
         self._add_current_stories_to_result()
         self.current_step_builder = StoryStepBuilder(name, source_name)
 
-    def _add_event(self, event_name: Text, parameters: Dict[Text, Any]) -> None:
+    def _add_event(
+        self, event_name: Text, parameters: Dict[Text, Any], is_e2e: bool = False
+    ) -> None:
 
         # add 'name' only if event is not a SlotSet,
         # because there might be a slot with slot_key='name'
         if "name" not in parameters and event_name != SlotSet.type_name:
-            parameters["name"] = event_name
+            # TODO: Hack by Genie as part of the prototype
```

Comment:
is it also related only to markdown support?

Reply:
No, I think this will work for both Markdown and Yaml. what it does in reality is clean up of the action text. (i'm not sure this is that much of a hack...)

Reply:
It's indeed only related to markdown support and should be removed

Reply:
done and removed!